nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/ExecuteScript.java

```diff
@@ -0,0 +1,585 @@
+     * to use other scripts, modules, etc. without having to build them into the ExecuteScript NAR.
+     * If the parameter is null or empty, ExecuteScript's classloader is returned
+     *
+     * @param modulePath The path to a directory containing modules to be used by the script(s)
+     */
+    private ClassLoader createScriptEngineModuleClassLoader(String modulePath) {
+        URLClassLoader newModuleClassLoader = null;
+        if (StringUtils.isEmpty(modulePath)) {
+            return ExecuteScript.class.getClassLoader();
+        }
+        try {
+            newModuleClassLoader =
+                    new URLClassLoader(
+                            new URL[]{new File(modulePath).toURI().toURL()}, ExecuteScript.class.getClassLoader());
+        } catch (MalformedURLException mue) {
+            getLogger().error("Couldn't find modules directory at " + modulePath, mue);
+        }
+        return newModuleClassLoader;
+    }
+
+    /**
+     * Evaluates the given script body (or file) using the current session, context, and flowfile. The script
+     * evaluation expects a FlowFile to be returned, in which case it will route the FlowFile to success. If a script
+     * error occurs, the original FlowFile will be routed to failure. If the script succeeds but does not return a
+     * FlowFile, the original FlowFile will be routed to no-flowfile
+     *
+     * @param context the current process context
+     * @param session the current process session
+     * @throws ProcessException if any error occurs during script evaluation
+     */
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        synchronized (isInitialized) {
+            if (!isInitialized.get()) {
+                createResources();
+            }
+        }
+        ProcessorLog log = getLogger();
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        String[] args = context.getProperty(SCRIPT_ARGS).evaluateAttributeExpressions().getValue().split(";");
+
+        Bindings bindings = scriptEngine.createBindings();
+
+        bindings.put("args", args);
+        bindings.put(ScriptEngine.ARGV, Arrays.asList(args));
+        bindings.put("session", session);
+        bindings.put("flowFile", flowFile);
+        bindings.put("log", log);
+
+        // Find the user-added properties and set them on the script
+        List<PropertyDescriptor> propertyDescriptors = getSupportedPropertyDescriptors();
+        for (Map.Entry<PropertyDescriptor, String> property : context.getProperties().entrySet()) {
+            if (!propertyDescriptors.contains(property.getKey())) {
+                // The descriptor isn't one of the supported ones, so it must be dynamic. Add it to the bindings
+                if (property.getValue() != null) {
+                    bindings.put(property.getKey().getName(), property.getValue());
```

Comment:
Instead of property.getValue(), I think this should be context.getProperty(property).evaluateAttributeExpressions(flowFile).getValue() so that we are evaluating the Expression Language before providing the value.